src/Apropos/Gen.hs

```diff
@@ -69,6 +99,8 @@ data FreeGen next where
     Gen a ->
     (a -> next) ->
     FreeGen next
+  Scale :: forall a next. (Int -> Int) -> Gen a -> (a -> next) -> FreeGen next
+  Prune :: forall a next. Gen a -> (a -> next) -> FreeGen next
```

Comment:
We can now `scale` and `prune` in `Gen`. We can add more shrink tree + size operations if need be.